stdlib/array.ml

```diff
@@ -185,6 +185,27 @@ let of_list = function
         | hd::tl -> unsafe_set a i hd; fill (i+1) tl in
       fill 1 tl
 
+let equal eq a b =
+  if length a <> length b then false else
+  try
+    for i = 0 to length a - 1 do
+      if eq (unsafe_get a i) (unsafe_get b i) then () else raise_notrace Exit
+    done;
+    true
+  with Exit -> false
```

Comment:
Can we avoid the exception? Something like below, or even a simple recursive function?
```suggestion
let equal eq a b =
  let len_a = length a and len_b = length b in
  let i = ref 0 in
  len_a = len_b && begin
    while !i < len_a && eq (unsafe_get a !i) (unsafe_get b !i) do
      incr i
    done;
    !i = len_a
  end
```

Reply:
Exception avoided in 11384582425bb4